discord/role.py

```diff
@@ -40,50 +41,24 @@ def __repr__(self):
 
 
 class Role(DiscordObject):
-    __slots__ = (
-        "_guild",
-        "_cache",
-        "_id",
-        "_name",
-        "_color",
-        "_hoist",
-        "_position",
-        "_permissions",
-        "_managed",
-        "_mentionable",
-        "_tags",
-    )
-
-    _guild: Guild
-    _cache: RoleCache
-    _id: Snowflake
-    _name: str
-    _color: Color
-    _hoist: bool
-    _position: int
-    _permissions: str
-    _managed: bool
-    _mentionable: bool
-    _tags: Optional[RoleTags]
-
-    def __init__(self, guild: Guild, cache: RoleCache, payload: RolePayload):
```

Comment:
How do we create a role from a payload if this is gone?

Reply:
why create it ourselves? pydantic will do it for us